app/javascript/angular/code/controllers/_sessions_list_ctrl.js

```diff
@@ -100,11 +100,11 @@ export const SessionsListCtrl = (
     var session = sessions.find(sessionId);
     if(sessions.isSelected(session)) {
       params.update({selectedSessionIds: []});
-      session.$selected = false;
+      elmApp.ports.toggleSessionSelection.send(null);
     } else if($scope.canSelectSession(sessionId)) {
       params.update({selectedSessionIds: [sessionId]});
       $scope.markerSelected.set(markerSelected);
-      session.$selected = true;
+      elmApp.ports.toggleSessionSelection.send(sessionId);
```

Comment:
We should investigate if we can get rid of the watch on line 111 at some point. As far as I can see it's responsible for calling `sessions.deselectSession` or `sessions.selecetSession` so maybe you can just call it from the `if else` directly?
That probably depends on how many other places mutate the `selectedSessionIds`, but I guess it shouldn't be mutated anywhere else so we can try to clean that up. It depends on how we will handle the use case when user change the sensors while having a selected session...

Reply:
Tried to remove but breaks some stuff. Maybe next round. Thanks for the comment!